### PR TITLE
fix: Smoke test codegen bug fix

### PR DESCRIPTION
**Pull request by sichanyoo**
## Description of changes

- Fix `orElse(null)` misuse
- Add ability to disable smoke test codegen by service name
- Fix bug where operations attached to the service's nested resource shape weren't getting their smoke tests generated properly.

## Scope

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/SmokeTestGenerator.kt [hunk 1]
```diff
@@ -16,25 +16,28 @@ import software.amazon.smithy.swift.codegen.utils.toUpperCamelCase
...
     private val ctx: ProtocolGenerator.GenerationContext
 ) {
-    // Filter out tests by name or tag at codegen time.
-    // Each element must have the prefix "<service-name>:" before the test name or tag name.
+    // Filter out tests by test name, tes tag, or service name at codegen time.
+    // Each test name or tag must have the prefix "<service-name>:" before the test name or tag name.
     // E.g., "ServiceX:ProcessOrderTest" or "ServiceX:Order"
     open val smokeTestIdsToIgnore = setOf<String>()
     open val smokeTestTagsToIgnore = setOf<String>()
+    open val servicesToIgnore = setOf<String>()
```
Comment (sichanyoo) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/SmokeTestGenerator.kt:24]: Add ability to disable smoke test codegen by service name

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/SmokeTestGenerator.kt [hunk 2]
```diff
@@ -53,10 +56,10 @@ open class SmokeTestGenerator(
      */
     private fun getOperationShapeIdToTestCasesMapping(serviceName: String): Map<ShapeId, List<SmokeTestCase>> {
         val operationShapeIdToTestCases = mutableMapOf<ShapeId, List<SmokeTestCase>>()
-        ctx.service.allOperations.forEach { op ->
-            if (ctx.model.expectShape(op).hasTrait<SmokeTestsTrait>()) {
+        ctx.model.operationShapes.forEach { op ->
```
Comment (sichanyoo) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/SmokeTestGenerator.kt:59]: Smoke test codegen was happening only for operations directly bound to service. The change here fixes the bug to codegen smoke tests even for operations bound to nested resource shapes of a service.

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/SmokeTestGenerator.kt [hunk 3]
```diff
@@ -202,7 +205,7 @@ open class SmokeTestGenerator(
         val clientName = getClientName()
         writer.write("let config = try await $clientName.${clientName}Configuration()")
         // Set any vendor-specific values into config.
-        handleVendorParams(testCase.vendorParams.orElse(null), writer)
+        handleVendorParams(testCase.vendorParams.orElse(ObjectNode.objectNode()), writer)
```
Comment (sichanyoo) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/SmokeTestGenerator.kt:208]: THIS was the cause of preview build failure xd

**Review by sichanyoo — COMMENTED** [review]
Comments for reviewers

**Review by jbelkins — APPROVED** [review]

**Pull request merged**